packages/common/src/compiler/exploreCompiler.ts

```diff
@@ -178,26 +178,24 @@ export class ExploreCompiler {
                                 requiredDimensionsForJoin.includes(
                                     dimensionKey,
                                 );
-                            const tableGroups =
-                                tables[join.table].groupDetails || {};
-                            const allGroupsIncluded =
+
+                            const isTimeIntervalBaseDimensionVisible =
+                                dimension.timeInterval &&
                                 dimension.groups &&
-                                dimension.groups.every(
-                                    (group) =>
-                                        join.fields !== undefined &&
-                                        (join.fields.includes(group) ||
-                                            join.fields.includes(
-                                                tableGroups[group].label,
```

Comment:
The problem was that `tableGroups[group]` might be undefined. 
We only need to check the last value from `dimension.groups` which is the dimension id of the base dimension from a time interval dimension.
Refactored logic.